### PR TITLE
Feature/367 remove titles

### LINKED ISSUE #367: Add functionality to remove titles
## ✨ Feature request
Is your feature request related to a problem? Please describe.

### 📇 User story
As a user I want to be able to remove a title so no more books can be assigned to that title

### 📜 Acceptance Criteria
A list of changes that need to be implemented to consider this feature done.
- [x] Button to remove existing title.
- [x] Removes title from DB
- [x] Generates warning if loan exist on a book connected to that title

### PR DESCRIPTION
**Pull request by te4-jonathan-ronsten**
## ⚙️ What issue did you implement or fix?
Fixes #367 

## 📜 Summary
- Added button to remove titles in edit mode 
- Added API calls in store to get active loans connected to a title
- Added support to generate warning if active loans exist
- Also made basic css changes in BaseModal component so \n works.

## 📝 Type of change

Please delete options that are not relevant.

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## 💩 How can this implementation be improved?
- Change bar into zar. :shipit:

### CODE REVIEW

**Review by te4-alexander-kjellberg — APPROVED** [review]
Looked through everything and it all looks good!

File: frontend/src/components/TitleFormComponent.vue [hunk 1]
```diff
@@ -1,92 +1,112 @@
...
+      v-if="show"
+      data-jest="form"
+      @submit.prevent="onSubmit"
+      @reset.prevent="onReset"
     >
-      <v-text-field
-        id="nestedBid"
-        v-model="form.name"
-        data-jest="name"
-        label="Namn"
```
Comment (Te4-Jacob-sandstrom) [inline frontend/src/components/TitleFormComponent.vue:17]: It is very hard to see if there are actually any changes to this file other than placing all elements inside main. Have you made any changes to the template?

**Review by Te4-Jacob-sandstrom — APPROVED** [review]
Looks good

File: frontend/src/components/TitleFormComponent.vue (hunk 1 above)
Comment (te4-jonathan-ronsten) [reply frontend/src/components/TitleFormComponent.vue:17]: Yes, i have added delete button if in 'edit mode' and also added the BaseModal component. Other than that i just placed everything in main

**Pull request merged**